dwave/plugins/torch/boltzmann_machine.py

```diff
@@ -214,31 +238,131 @@ def interactions(self, x: torch.Tensor) -> torch.Tensor:
         """
         return x[..., self.edge_idx_i] * x[..., self.edge_idx_j]
 
-    def sufficient_statistics(
-        self, x: torch.Tensor
-    ) -> tuple[torch.Tensor, torch.Tensor]:
-        """Compute the sufficient statistics of a Boltzmann machine, i.e., average spin
-        and average interaction values (per edge) of ``x``.
+    def sufficient_statistics(self, x: torch.Tensor) -> torch.Tensor:
+        """Compute the sufficient statistics of a Boltzmann machine, i.e., spins
+        concatenated with interaction values (per edge) of ``x``.
 
         Args:
             x (torch.Tensor): A tensor of shape (..., N) where N denotes the number of
                 variables in the model.
 
         Returns:
-            tuple[torch.Tensor, torch.Tensor]: The sufficient statistics of ``x``.
+            torch.Tensor: The sufficient statistics of ``x``.
         """
         interactions = self.interactions(x)
-        return x.mean(dim=0), interactions.mean(dim=0)
+        return torch.cat([x, interactions], 1)
 
-    @property
-    def _ising(self) -> tuple[dict, dict]:
+    def _ising(self, prefactor: float, h_range: tuple[float, float]=None, j_range: tuple[float, float]=None) -> tuple[dict, dict]:
         """Convert the model to Ising format"""
-        linear_biases = self.h.detach().cpu().tolist()
+        h = prefactor * self.h.detach()
+        J = prefactor * self.J.detach()
+        if h_range is not None:
+            h = h.clip(*h_range)
+        if j_range is not None:
+            J = J.clip(*j_range)
+
         edge_idx_i = self.edge_idx_i.detach().cpu().tolist()
         edge_idx_j = self.edge_idx_j.detach().cpu().tolist()
-        quadratic_bias_list = self.J.detach().cpu().tolist()
-        quadratic_biases = {
-            (a, b): w for a, b, w in zip(edge_idx_i, edge_idx_j, quadratic_bias_list)
+        h = {self.idx_to_var[i]: b for i, b in enumerate(h.cpu().tolist())}
+        J = {
+            (self.idx_to_var[a], self.idx_to_var[b]): w
+            for a, b, w in zip(edge_idx_i, edge_idx_j, J)
         }
+        return h, J
+
+    @property
+    def ising(self):
+        return self._ising(1)
+
+    @property
+    def bqm(self):
+        bqm = BinaryQuadraticModel.from_ising(*self.ising)
```

Comment:
Missing docstring. I wonder if this is necessary at all though. Couldn't we just let the user create a BQM themselves if needed. I guess we'd still need to import BQM from dimod to use it for the temperature estimation though.

Reply:
(removed this method)